### ISSUE TITLE
`argo auth token` output is invalid on first try, but works on re-run

### ISSUE THREAD

**Issue by danxmoran** [argoproj/argo#2310]
Checklist:

 * [x] I've included the version.
 * [x] I've included reproduction steps.
 * [ ] I've included the workflow YAML.
 * [ ] I've included the logs.
 
**What happened**:
My team recently deployed the Argo server in hosted mode, using the "client" mode for auth. I've noticed the same pattern a few times:
1. I navigate to the server's URL, and get redirected to the login page
2. I run `argo auth token` to generate a login token. The command takes a few seconds to execute (I assume because it's renewing a locally-cached token), then prints `Bearer <token>`
3. I input `Bearer <token>` and click "login", but get redirected back to the login page
4. I re-run `argo auth token`. The command outputs immediately, _but_ the output is some different `Bearer <token2>`
5. I input `Bearer <token2>` and click "login", and get redirected to the workflows page

**What you expected to happen**:
I expect every token produced by `argo auth token` to be valid.

**How to reproduce it (as minimally and precisely as possible)**:
I'm not sure how general of a problem this is. On my machine, I can reproduce as easily as running `argo auth token` 2 times in a row after around an hour-long pause (to let the cached token expire).

**Anything else we need to know?**:
Based purely on the timing of how the commands are returning, could it be that `argo auth token` is doing something like:
1. `token` := Read locally-cached token
2. If `token` is expired, renew and write to local cache
3. Print `token`

And the bug is that step 3 should print the new value, if one was fetched?

**Environment**:
- Argo version:
```
$ argo version
argo: v2.5.1
  BuildDate: 2020-02-20T18:20:18Z
  GitCommit: fb496a244383822af5d4c71431062cebd6de0ee4
  GitTreeState: clean
  GitTag: v2.5.1
  GoVersion: go1.13.4
  Compiler: gc
  Platform: darwin/amd64
```
---

**Message from the maintainers**:

If you are impacted by this bug please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by danxmoran** [comment]
More detail: I am running against GCP / GKE, using the `gcp` auth provider. I found [this branch](https://github.com/argoproj/argo/blob/master/util/kubeconfig/kubeconfig.go#L147-L169) in the auth code, which I think support my theory. `in.AuthProvider.Config` is cached from when the CLP boots up, so even if `rt.RoundTrip(&req)` refreshes the token the new value isn't returned.

**Comment by simster7** [comment]
@sarabala1979 Can you take a look at this?

**Comment by simster7** [comment]
Fixed by #2327
